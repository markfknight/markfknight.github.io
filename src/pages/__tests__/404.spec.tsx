import React from 'react';
import { cleanup, render } from '@testing-library/react';

import NotFoundPage, { Data } from '../404';
import { PageProps } from 'gatsby';

jest.mock('../../components/seo');

afterEach(cleanup);

describe(`404 snapshot`, () => {
  it(`renders correclty`, () => {
    const props: PageProps<Data> = {
      location: {},
      data: {
        site: {
          siteMetadata: {
            title: 'test',
          },
        },
      },
    } as PageProps<Data>;

    const tree = render(<NotFoundPage {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
