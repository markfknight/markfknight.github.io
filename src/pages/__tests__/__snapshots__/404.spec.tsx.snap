// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 snapshot renders correclty 1`] = `
<div>
  <div
    class="container max-w-3xl px4 py10"
  >
    <header
      class="py-10"
    >
      <h3>
        <a
          href="/"
        >
          test
        </a>
      </h3>
    </header>
    <main>
      <h1>
        Not Found
      </h1>
      <p>
        You just hit a route that doesn't exist... the sadness.
      </p>
    </main>
    <footer
      class="py-10 text-center"
    >
      Â© 
      2021
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>
</div>
`;
